File: scripts/SecondLargest.java
import java.util.Scanner;


class SecondLargest
{
	public static void main(String args[])
	{
		int a,b,c;
		int second=0;
		a = 100; b = 20; c = 30;
		if(a>b && a>c)
		{	
			if (b>c)
			{
				second = b;
			}
			if (c>b)
			{
				second = c;
			}
		}
		else if(b>a && b>c)
		{
			if (a>c)
			{
				second = a;
			}
			if (c>a)
			{
				second = c;
			}
		}
		else
		{
			if (b>a)
			{
				second = b;
			}
			if (a>b)
			{
				second = a;
			}
		}
		System.out.println("Second Largest is "+second);
		
	} 
	

}

File: Program's_Contributed_By_Contributors/Golang Programs/secondLargest.go
package main

import "fmt"

func findSecondLargest(numbers []int) int {
    if len(numbers) < 2 {
        return -1 // Not enough elements to determine the second largest.
    }

    largest, secondLargest := numbers[0], numbers[0]

    for _, num := range numbers {
        if num > largest {
            secondLargest = largest
            largest = num
        } else if num > secondLargest && num < largest {
            secondLargest = num
        }
    }

    return secondLargest
}

func main() {
    numbers := []int{10, 5, 8, 20, 15, 10, 30}
    secondLargest := findSecondLargest(numbers)
    fmt.Printf("The second largest number in the list is: %d\n", secondLargest)
}
